ding/policy/prompt_dt.py

```diff
@@ -0,0 +1,301 @@
+from typing import List, Dict, Any, Tuple, Optional
+from collections import namedtuple
+import torch.nn.functional as F
+import torch
+import numpy as np
+from ding.torch_utils import to_device
+from ding.utils import POLICY_REGISTRY
+from ding.utils.data import default_decollate
+from ding.policy.dt import DTPolicy
+from ding.model import model_wrap
+
+@POLICY_REGISTRY.register('promptdt')
+class PDTPolicy(DTPolicy):
+    """
+    Overview:
+        Policy class of Decision Transformer algorithm in discrete environments.
+        Paper link: https://arxiv.org/pdf/2206.13499.
+    """
+    def default_model(self) -> Tuple[str, List[str]]:
+        return 'dt', ['ding.model.template.decision_transformer']
+    
+    def _init_learn(self) -> None:
+        super()._init_learn()
+        self.need_prompt = self._cfg.need_prompt
+
+    def _forward_learn(self, data: List[torch.Tensor]) -> Dict[str, Any]:
+        """
+        Overview:
+            Policy forward function of learn mode (training policy and updating parameters). Forward means \
+            that the policy inputs some training batch data from the offline dataset and then returns the output \
+            result, including various training information such as loss, current learning rate.
+        Arguments:
+            - data (:obj:`List[torch.Tensor]`): The input data used for policy forward, including a series of \
+                processed torch.Tensor data, i.e., timesteps, states, actions, returns_to_go, traj_mask.
+        Returns:
+            - info_dict (:obj:`Dict[str, Any]`): The information dict that indicated training result, which will be \
+                recorded in text log and tensorboard, values must be python scalar or a list of scalars. For the \
+                detailed definition of the dict, refer to the code of ``_monitor_vars_learn`` method.
+
+        .. note::
+            The input value can be torch.Tensor or dict/list combinations and current policy supports all of them. \
+            For the data type that not supported, the main reason is that the corresponding model does not support it. \
+            You can implement you own model rather than use the default model. For more information, please raise an \
+            issue in GitHub repo and we will continue to follow up.
+
+        """
+        self._learn_model.train()
+        self.have_train = True
+
+        if self._cuda:
+            data = to_device(data, self._device)
+
+        p_s, p_a, p_rtg, p_t, p_mask, timesteps, states, actions, rewards, returns_to_go, \
```

Comment:
data should be collated into batchsize before entering policy._forward_learn.
data type shoule be Dict[str, torch.Tensor], so that it can be assigned confirmly.